codec/src/main/java/io/netty/handler/codec/string/LineSeparator.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.string;
+
+import io.netty.util.internal.ObjectUtil;
+import io.netty.util.internal.StringUtil;
+
+/**
+ * A class to represent line separators in different environments.
+ */
+public final class LineSeparator {
+
+    /**
+     * The default line separator in the current system.
+     */
+    public static final LineSeparator DEFAULT = new LineSeparator(StringUtil.NEWLINE);
+
+    /**
+     * The Unix line separator(LF)
+     */
+    public static final LineSeparator UNIX = new LineSeparator("\n");
+
+    /**
+     * The Windows line separator(CRLF)
+     */
+    public static final LineSeparator WINDOWS = new LineSeparator("\r\n");
+
+    private final String value;
+
+    /**
+     * Create {@link LineSeparator} with the specified {@code lineSeparator} string.
+     */
+    public LineSeparator(String lineSeparator) {
+        ObjectUtil.checkNotNull(lineSeparator, "lineSeparator");
+        this.value = lineSeparator;
+    }
+
+    /**
+     * Return the string value of this line separator.
+     */
+    public String value() {
+        return value;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || !(o instanceof LineSeparator)) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) No need to check for null before an instanceof [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3ASimplifyConditional)